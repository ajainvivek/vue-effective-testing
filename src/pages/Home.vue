<template>
    <div class="Home">
        <list :list="list" :loading="loading" @fetchMore="loadMore"></list>
    </div>    
</template>

<script>
import List from '@/components/List.vue'

export default {
    name: 'home',
    components: {
       List 
    },
    data () {
        return {
            loading: false,
            list: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
        };
    },
    methods: {
        loadMore() {
            this.loading = true;
            setTimeout(() => {
                let last = this.list[this.list.length - 1];
                for (let i = 1; i <= 10; i++) {
                    this.list.push(last + i);
                }
                this.loading = false;
            }, 2500);
        }
    }
}
</script>
