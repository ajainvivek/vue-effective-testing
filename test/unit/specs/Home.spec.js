import { shallow } from '@vue/test-utils'
import Home from '@/pages/Home.vue'
import { mockCharacters } from './__mocks__/home.data';

describe('Home.vue', () => {
  
    let page = null;
  
    beforeEach(() => {
        page = shallow(Home)
    });
  
    it('should have mounted', () => {
        expect(page.element.className).toEqual("Home");
    });
  
    it('matches snapshot', () => {
      expect(page.html()).toMatchSnapshot();
    });
  })
  