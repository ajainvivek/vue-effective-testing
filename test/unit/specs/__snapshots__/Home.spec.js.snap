// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue matches snapshot 1`] = `
<div class="Home">
    <!---->
</div>
`;
